File: silk-core/src/main/scala/org/silkframework/runtime/plugin/annotations/PluginReference.java
package org.silkframework.runtime.plugin.annotations;

import java.lang.annotation.*;

/**
 * Reference to another plugin.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface PluginReference {

    /** The unique id of the referenced plugin */
    String id();

    /** An optional short description of the relationship to the referenced plugin. */
    String description() default "";
}

File: silk-core/src/main/scala/org/silkframework/runtime/plugin/PluginReference.scala
package org.silkframework.runtime.plugin

import org.silkframework.util.Identifier

/**
 * Reference to a plugin.
 *
 * @param id The identifier of the referenced plugin.
 * @param description An optional description of the relationship to the referenced plugin.
 */
case class PluginReference(id: Identifier, description: Option[String])
